connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -357,7 +357,53 @@ object functions {
   def collect_list(columnName: String): Column = collect_list(Column(columnName))
 
   /**
-   * Aggregate function: returns a set of objects with duplicate elements eliminated.
+   * Aggregate function: returns a list of objects with duplicates.
+   *
+   * The parameter ignoreNulls controls if nulls should be excluded from the result.
+   *
+   * @note
+   *   The function is non-deterministic because the order of collected results depends on the
+   *   order of the rows which may be non-deterministic after a shuffle.
+   *
+   * @group agg_funcs
+   * @since 4.0.0
+   */
+  def collect_list(e: Column, ignoreNulls: Boolean): Column =
+    collect_list(e, lit(ignoreNulls))
+
+  /**
+   * Aggregate function: returns a list of objects with duplicates.
+   *
+   * The parameter ignoreNulls controls if nulls should be excluded from the result.
+   *
+   * @note
+   *   The function is non-deterministic because the order of collected results depends on the
+   *   order of the rows which may be non-deterministic after a shuffle.
+   *
+   * @group agg_funcs
+   * @since 4.0.0
+   */
+  def collect_list(e: Column, ignoreNulls: Column): Column =
```

Comment:
If this works for non-foldable expressions too, I would remove `ignoreNulls: Boolean` signatures (see the top of this file)

Reply:
I does not work for non-foldable expressions. I based it on how it currently works for `First` and `Last` here https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/First.scala#L123-L127 so basically it only supports literal boolean. But I guess it makes sense to change that to be any foldable expression returning a boolean.

My reading of the comment is that maybe we should still remove the overload taking a `Boolean`? Does that also apply for the scala API or only the connect api?

Reply:
Hm .. let's still just have `Column` signature for now. Suppose that we want to support non-foldable expression here, we will have to add another method if we add `Boolean` here. Would be great if we canavoid adding too many overloaded functions.

Reply:
Thanks for the feedback. I made the suggested changes.